packages/interactivity/src/proxies/test/state-proxy.ts

```diff
@@ -0,0 +1,1187 @@
+
+/**
+ * External dependencies
+ */
+import { effect } from '@preact/signals-core';
+/**
+ * Internal dependencies
+ */
+import { proxifyState, peek } from '../';
+import {
+	setScope,
+	resetScope,
+	setNamespace,
+	resetNamespace,
+	getContext,
+	getElement,
+} from '../../hooks';
+
+type State = {
+	a?: number;
+	nested: { b?: number };
+	array: ( number | State[ 'nested' ] )[];
+};
+
+const withScopeAndNs = ( scope, ns, callback ) => () => {
+	setScope( scope );
+	setNamespace( ns );
+	try {
+		return callback();
+	} finally {
+		resetNamespace();
+		resetScope();
+	}
+};
+
+describe( 'interactivity api - state proxy', () => {
+	let nested = { b: 2 };
+	let array = [ 3, nested ];
+	let raw: State = { a: 1, nested, array };
+	let state = proxifyState( 'test', raw );
+
+	const window = globalThis as any;
+
+	beforeEach( () => {
+		nested = { b: 2 };
+		array = [ 3, nested ];
+		raw = { a: 1, nested, array };
+		state = proxifyState( 'test', raw );
+	} );
+
+	describe( 'get - plain', () => {
+		it( 'should return plain objects/arrays', () => {
+			expect( state.nested ).toEqual( { b: 2 } );
+			expect( state.array ).toEqual( [ 3, { b: 2 } ] );
+			expect( state.array[ 1 ] ).toEqual( { b: 2 } );
+		} );
+
+		it( 'should return plain primitives', () => {
+			expect( state.a ).toBe( 1 );
+			expect( state.nested.b ).toBe( 2 );
+			expect( state.array[ 0 ] ).toBe( 3 );
+			expect(
+				typeof state.array[ 1 ] === 'object' && state.array[ 1 ].b
+			).toBe( 2 );
+			expect( state.array.length ).toBe( 2 );
+		} );
+
+		it( 'should support reading from getters', () => {
+			const state = proxifyState( 'test', {
+				counter: 1,
+				get double() {
+					return state.counter * 2;
+				},
+			} );
+			expect( state.double ).toBe( 2 );
+			state.counter = 2;
+			expect( state.double ).toBe( 4 );
+		} );
+
+		it( 'should support getters returning other parts of the state', () => {
+			const state = proxifyState( 'test', {
+				switch: 'a',
+				a: { data: 'a' },
+				b: { data: 'b' },
+				get aOrB() {
+					return state.switch === 'a' ? state.a : state.b;
+				},
+			} );
+			expect( state.aOrB.data ).toBe( 'a' );
+			state.switch = 'b';
+			expect( state.aOrB.data ).toBe( 'b' );
+		} );
+
+		it( 'should support getters using ownKeys traps', () => {
+			const state = proxifyState( 'test', {
+				x: {
+					a: 1,
+					b: 2,
+				},
+				get y() {
+					return Object.values( state.x );
+				},
+			} );
+
+			expect( state.y ).toEqual( [ 1, 2 ] );
+		} );
+
+		it( 'should support getters accessing the scope', () => {
+			const state = proxifyState( 'test', {
+				get y() {
+					const ctx = getContext< { value: string } >();
+					return ctx.value;
+				},
+			} );
+
+			const scope = { context: { test: { value: 'from context' } } };
+			try {
+				setScope( scope as any );
+				setNamespace( 'test' );
+				expect( state.y ).toBe( 'from context' );
+			} finally {
+				resetNamespace();
+				resetScope();
+			}
```

Comment:
Shouldn't namespace be interally set?

The test pass with this:

```suggestion
			try {
				setScope( scope as any );
				expect( state.y ).toBe( 'from context' );
			} finally {
				resetScope();
			}
```

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/62734/commits/8a6589aba4a69b7f339a327ffa3543f2b3ea8ef0